src/utils.rs

```diff
@@ -1,9 +1,27 @@
 //! Utility date and time equations for Temporal
 
+use std::cmp::Ordering;
+
 use crate::MS_PER_DAY;
 
 // NOTE: Review the below for optimizations and add ALOT of tests.
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct NonZeroOrdering(Ordering);
+
+impl NonZeroOrdering {
+    pub(crate) fn new(ordering: Ordering) -> Self {
+        if matches!(ordering, Ordering::Less) {
+            return Self(ordering);
+        }
+        Self(Ordering::Greater)
+    }
+
+    pub(crate) fn get(&self) -> i8 {
+        self.0 as i8
+    }
+}
```

Comment:
Seeing that this is only used by `sign`, maybe it can just be a wrapper around `NonZeroI8`?
```rust
struct Sign(NonZeroI8);

impl Sign {
    const POSITIVE: Self = const {
        let Some(n) = NonZeroI8::new(1) else {
            panic!("invalid number for sign");
        }
        Self(n)
    }

    const NEGATIVE: Self = const {
        let Some(n) = NonZeroI8::new(-1) else {
            panic!("invalid number for sign");
        }
        Self(n)
    }

    pub(crate) fn get(&self) -> i8 {
        self.0.get()
    }
}
```
This also makes `std::mem::size_of::<Option<Sign>>() == std::mem::size_of::<i8>()`

Reply:
Actually, an `enum Sign` optimizes perfectly fine, apparently.
```rust
enum Sign {
    Positive = 1,
    Negative = -1,
}

dbg!(std::mem::size_of::<Sign>()); // 1
dbg!(std::mem::size_of::<Option<Sign>>()); // 1
```

Reply:
Hmmm maybe a Sign, which can be zero, and a NonZeroSign (hence the NonZeroOrdering).

Reply:
Do we need a sign that can be zero though? I would expect it to only have Positive and Negative, since the zero case is not really useful IIRC.

Reply:
Temporal Spec has a -1, 0, 1 return

https://tc39.es/proposal-temporal/#sec-temporal-durationsign